src/kernel/src/process/vm/mod.rs

```diff
@@ -838,3 +969,43 @@ pub enum MemoryUpdateError {
     #[error("address {0:p} is not mapped")]
     UnmappedAddr(*const u8),
 }
+
+#[repr(C)]
+struct BatchMapOp {
+    addr: usize,
+    offset: usize,
+    len: usize,
+    prot: u8,
+    ty: u8,
+    op: MapOp,
```

Comment:
I think using an enum here causes potentially undefined behavirour when you create a *BatchMapOp from a SysArg.

Reply:
Is using i32 better?

Reply:
It depends on how many bytes the PS4 reads. If 4, then yes.

Reply:
Seems to read 4, I wrote it a bit different